src/System.Text.RegularExpressions/tests/RegexParserTests.cs

```diff
@@ -0,0 +1,1095 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using Xunit;
+using Xunit.Sdk;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    public class RegexParserTests
+    {
+        private static readonly Type s_parseExceptionType;
+        private static readonly FieldInfo s_parseErrorField;
+
+        static RegexParserTests()
+        {
+            // On Full Framework RegexParseException doesn't exist and on uapaot reflection is blocked.
+            if (!PlatformDetection.IsNetNative && !PlatformDetection.IsFullFramework)
+            {
+                s_parseExceptionType = typeof(Regex).Assembly.GetType("System.Text.RegularExpressions.RegexParseException", true);
+                s_parseErrorField = s_parseExceptionType.GetField("_error", BindingFlags.NonPublic | BindingFlags.Instance);
+            }
+        }
+
+        [Theory]
+        // Basic
+        [InlineData(@"""""", RegexOptions.None, null, true)]
+        [InlineData(@""" """, RegexOptions.IgnorePatternWhitespace, null, true)]
+        [InlineData(@"""  """, RegexOptions.IgnorePatternWhitespace, null, true)]
+        [InlineData(@"""(?#)""", RegexOptions.None, null, true)]
+        [InlineData(@"""(?# )""", RegexOptions.None, null, true)]
+        [InlineData(@"""(?#""", RegexOptions.None, RegexParseError.UnterminatedComment, true)]
+        [InlineData(@"""(?# """, RegexOptions.None, RegexParseError.UnterminatedComment, true)]
+        [InlineData(@"""(?#)(?#)""", RegexOptions.None, null, true)]
+        [InlineData(@"""(?#)(?#)""", RegexOptions.IgnorePatternWhitespace, null, true)]
+        [InlineData(@"""(?#) (?#)""", RegexOptions.None, null, true)]
+        [InlineData(@"""(?#) (?#)""", RegexOptions.IgnorePatternWhitespace, null, true)]
+        [InlineData(@"@""[a\p{Lu}(?#)b]""", RegexOptions.None, null, true)]
```

Comment:
What are the nested @ for? I wasn't aware it was special in regexes - but maybe I'm wrong as I see `char ch = '@'; // nonspecial ch, means at beginning` in the code.

Reply:
The nested ```@""...""``` should be removed.   This is a roslyn-test-ism since i'm first generating a C# string token, since the entire Roslyn embedded-regex stack starts with a String-token which then gets analyzed.  I probably should have written my tests to not have that code and to instead just have the regex contents . I could have wrapped those tests in the test harness with ```@"..."``` instead.  But i wasn't thinking about these being reused here.

Reply:
@CyrusNajmabadi now I feel bad that you have to explain several times. When Dan asked I was nearly done changing the test data 😄

Reply:
I feel bad i put you through this!  It was a silly shortcut i didn't think about when writing these tests.

Reply:
Don't feel bad :) You helped us a lot with the test cases. And I know that you are waiting for a response in the other thread, I'll answer as soon as possible.